test/battle/ability/magic_guard.c

```diff
@@ -0,0 +1,16 @@
+#include "global.h"
+#include "test/battle.h"
+
+SINGLE_BATTLE_TEST("Magic Guard prevents recoil damage to the user")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_DOUBLE_EDGE].recoil == 33);
+        PLAYER(SPECIES_CLEFABLE) { Ability(ABILITY_MAGIC_GUARD); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_DOUBLE_EDGE); }
+    } SCENE {
+        NOT HP_BAR(player);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_DOUBLE_EDGE, player);
```

Comment:
the `NOT` should come after the animation. Also I would add an HP_BAR for the opponent so:

```
        ANIMATION(ANIM_TYPE_MOVE, MOVE_DOUBLE_EDGE, player);
        HP_BAR(opponent);
        NOT HP_BAR(player);

```

Reply:
I'm fine with this change, but there are very similar tests in `max_hp_50_recoil.c` and `mind_blown.c` which will now be inconsistently formatted with what we're doing here. The format you proposed is probably more correct and I don't want to not do the correct thing just because something less correct is the precedent.

Reply:
This is not a formatting issue but rather a wrongly written test. Recoil is taken after the move animation not before. That just means it is wrongly written in those files as well. If you want you can correct them 😅

Reply:
Gotcha, they've been corrected.

Reply:
Thx!